File: Module.Frontend.TPM/Client/model/tpm/clienttree/ClientTree.js
Ext.define('App.model.tpm.clienttree.ClientTree', {
    extend: 'Ext.data.Model',
    idProperty: 'ObjectId',
    breezeEntityType: 'ClientTree',
    fields: [
        { name: 'Id', type: 'int', hidden: true },
        { name: 'ObjectId', hidden: true },
        { name: 'depth', type: 'int', persist: true },
        { name: 'Type', type: 'string', hidden: false, isDefault: true },
        { name: 'RetailTypeName', type: 'string', hidden: false, isDefault: true },
        { name: 'Name', type: 'string', hidden: false, isDefault: true },
        { name: 'FullPathName', type: 'string', hidden: false, isDefault: true },
        { name: 'StartDate', type: 'date', hidden: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: 'EndDate', type: 'date', hidden: true, useNull: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: 'leaf', type: 'bool', defaultValue: false, persist: false },
        { name: '_checked', type: 'bool', defaultValue: false, persist: false },
        { name: 'IsBaseClient', type: 'bool', defaultValue: false },
        { name: 'GHierarchyCode', type: 'string', hidden: false, isDefault: true },
        { name: 'DemandCode', type: 'string', hidden: false, isDefault: true },
        { name: 'IsBeforeStart', type: 'bool', hidden: false, useNull: true },
        { name: 'IsDaysStart', type: 'bool', hidden: false, useNull: true },
        { name: 'DaysStart', type: 'int', hidden: false, useNull: true },
        { name: 'IsBeforeEnd', type: 'bool', hidden: false, useNull: true },
        { name: 'IsDaysEnd', type: 'bool', hidden: false, useNull: true },
        { name: 'DaysEnd', type: 'int', hidden: false, useNull: true },
        { name: 'PostPromoEffectW1', type: 'float', hidden: false, useNull: true, defaultValue: 0 },
        { name: 'PostPromoEffectW2', type: 'float', hidden: false, useNull: true, defaultValue: 0 },
        { name: 'LogoFileName', type: 'string', hidden: false, useNull: true, defaultValue: null },
        { name: 'IsOnInvoice', type: 'bool', useNull: true, defaultValue: null },
        { name: 'DMDGroup', type: 'string', hidden: false, isDefault: true  },
        {
            name: 'DeviationCoefficient', type: 'float', hidden: false, isDefault: true,
            convert: function (value) {
                if (value === null)
                    value = 0;

                return value * 100;
            }
        },
        { name: 'DistrMarkUp', type: 'float', hidden: false, useNull: true, defaultValue: null },
        { name: 'SFAClientCode', type: 'string', hidden: false, useNull: true, defaultValue: null },
    ],

    proxy: {
        type: 'ajax',
        resourceName: 'ClientTrees',
        api: {
            create: '/odata/ClientTrees',
            read: '/odata/ClientTrees',
            update: '/odata/ClientTrees/UpdateNode',
            //destroy: 'destroyPersons'
        },
        extraParams: {
            filterParameter: null,
            clientObjectId: null,
            budgetSubItemId: null,
            needBaseClients: false,
            dateFilter: null,
            view: false,
        }
    }
});

File: Module.Persist.TPM/Model/TPM/ClientTree.cs
using Core.Data;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Module.Persist.TPM.Model.TPM
{
    public class ClientTree : IEntity<int>, ICloneable
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        [Index("CX_ObjDate", 1, IsUnique = true)]
        public int ObjectId { get; set; }
        public int parentId { get; set; }
        public int depth { get; set; }
        public string Type { get; set; }
        public string RetailTypeName { get; set; }
        public string Name { get; set; }
        public string FullPathName { get; set; }
        public bool? IsOnInvoice { get; set; }
        public string DMDGroup { get; set; }
        public DateTime StartDate { get; set; }
        [Index("CX_ObjDate", 2, IsUnique = true)]
        public DateTime? EndDate { get; set; }
        [StringLength(255)]
        public string GHierarchyCode { get; set; }
        [StringLength(255)]
        public string DemandCode { get; set; }
        public bool IsBaseClient { get; set; }

        //Dispatch start
        public bool? IsBeforeStart { get; set; }
        public int? DaysStart { get; set; }
        public bool? IsDaysStart { get; set; }

        //Dispatch end
        public bool? IsBeforeEnd { get; set; }
        public int? DaysEnd { get; set; }
        public bool? IsDaysEnd { get; set; }

        public double? DeviationCoefficient { get; set; }

        public string LogoFileName { get; set; }

        public double? DistrMarkUp { get; set; }

        public string SFAClientCode { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTimeOffset? ModifiedDate { get; set; }

        public ICollection<AssortmentMatrix> AssortmentMatrices { get; set; }
        public ICollection<BudgetSubItemClientTree> BudgetSubItemClientTrees { get; set; }
        public ICollection<ClientTreeBrandTech> ClientTreeBrandTeches { get; set; }
        public ICollection<CompetitorPromo> CompetitorPromoes { get; set; }
        public ICollection<EventClientTree> EventClientTrees { get; set; }
        public ICollection<MechanicType> MechanicTypes { get; set; }
        public ICollection<NoneNego> NoneNegoes { get; set; }
        public ICollection<NonPromoSupport> NonPromoSupports { get; set; }
        public ICollection<Plu> Plus { get; set; }
        public ICollection<PriceList> PriceLists { get; set; }
        public ICollection<PromoSupport> PromoSupports { get; set; }
        public ICollection<RATIShopper> RATIShoppers { get; set; }
        public ICollection<TradeInvestment> TradeInvestments { get; set; }
        public ICollection<PlanPostPromoEffect> PlanPostPromoEffects { get; set; }
        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}
